### PR TITLE
#2222 reconnect confirmation page

### PR DESCRIPTION
**Pull request by chrispymm**
Updates on [ticket #2222](https://trello.com/c/awSx6IDL/2222-adding-pages-at-the-end-of-the-flow-when-cya-conf-are-not-present-s)

 - 'Add confirmation page' should only be present in the connection menu after a cya page, if there is no unconnected confirmation page, else it would be 'reconnect confirmation page'

 - All other connection menus would never contain 'Add confirmation page'

 - Other connection menus could contain 'Add check your answers page' if one is not already present in the form (connected or unconnected)

 - As an extra bonus, this should now only show the cya page as an option if you are at the end of the flow.

image

image

### CODE REVIEW

File: app/views/services/_connection_menu.html.erb [hunk 1]
```diff
@@ -36,26 +36,24 @@
...
-            data-destination-uuid="<%= service.confirmation_page.uuid -%>">
-          <a href="#reconnect-confirmation">Reconnect confirmation page</a>
-        </li>
+    <% if item[:type] == 'page.checkanswers' %>
+      <% if service.confirmation_page.present? %>
+        <% if !@publish_warning.confirmation_in_main_flow? %>
+          <li data-action="reconnect-confirmation"
+              data-url="<%= api_service_flow_destinations_path(service.service_id, item[:uuid]) -%>"
+              data-destination-uuid="<%= service.confirmation_page.uuid -%>">
+            <a href="#reconnect-confirmation">Reconnect confirmation page</a>
```
Comment (brenetic) [inline app/views/services/_connection_menu.html.erb:49]: can we add the `Reconnect confirmation page` to the locales please

File: app/views/services/_connection_menu.html.erb [hunk 2]
```diff
@@ -36,26 +36,24 @@
...
+          </li>
+        <% end %>
+      <% else %>
+        <li data-page-type="confirmation"><a href="#add-page">Add confirmation page</a></li>
```
Comment (brenetic) [inline app/views/services/_connection_menu.html.erb:53]: can we add the `Add confirmation page` to the locales please

File: app/views/services/_connection_menu.html.erb [hunk 3]
```diff
@@ -36,26 +36,24 @@
         <li data-page-type="exit"><a href="#add-page">Add exit page</a></li>
     <% end %>
 
-    <% unless service.checkanswers_page.present? %>
+    <% if !service.checkanswers_page.present? && item[:next]&.empty?  %>
```
[the remark this answers is not in the document]
Comment (chrispymm) [reply app/views/services/_connection_menu.html.erb]: yep, did my research this morning and fixed it better!

**Comment by chrispymm** [comment]
Closing this PR as I've made further changes in a testable branch, so will PR from that branch instead.  Sorry for the confusion

**Pull request closed without merging**